python/lsst/ts/externalscripts/make_love_stress_tests.py

```diff
@@ -0,0 +1,324 @@
+        return yaml.safe_load(schema_yaml)
+
+    def set_metadata(self, metadata):
+        """Compute estimated duration.
+
+        Parameters
+        ----------
+        metadata : SAPY_Script.Script_logevent_metadataC
+        """
+        # a crude estimate;
+        metadata.duration = len(self.clients) * 5
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Specify the Stress test configurations:
+        - LOVE host location
+        - Number of clients
+        - Number of messages to store
+        - CSCs
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration with several attributes, defined in `get_schema`
+
+        Notes
+        -----
+        Saves the results on several attributes:
+
+        * config    : `types.SimpleNamespace`, same as config param
+        * remotes   : a dict, with each item as
+            CSC_name[:index]: `lsst.ts.salobj.Remote`
+
+        Constructing a `salobj.Remote` is slow (DM-17904), so configuration
+        may take a 10s or 100s of seconds per CSC.
+        """
+        self.log.info("Configure started")
+
+        # set configurations
+        self.config = config
+
+        # get credentials
+        self.username = "user"
+        self.password = os.environ.get("USER_USER_PASS")
+        if self.password is None:
+            raise RuntimeError(
+                "Configuration failed: environment variable USER_USER_PASS not defined"
+            )
+
+        # construct remotes
+        remotes = dict()
+        for name_index in config.data:
+            name, index = salobj.name_to_name_index(name_index)
+            self.log.debug(f"Create remote {name}:{index}")
+            if name_index not in remotes:
+                remote = salobj.Remote(domain=self.domain, name=name, index=index)
+                remotes[name_index] = remote
+        self.remotes = remotes
+
+    async def run(self):
+        """Run script."""
+
+        self.log.info(f"Waiting for {len(self.remotes)} remotes to be ready")
+        await asyncio.gather(*[remote.start_task for remote in self.remotes.values()])
+
+        # Enable all CSCs
+        for remote in self.remotes.values():
+            await salobj.set_summary_state(remote=remote, state=salobj.State.ENABLED)
+
+        event_streams = dict()
+        telemetry_streams = dict()
+        for name_index in self.remotes:
+            name, index = salobj.name_to_name_index(name_index)
+            salinfo = salobj.SalInfo(self.domain, name)
+            try:
+                event_streams[name_index] = salinfo.event_names
+                telemetry_streams[name_index] = salinfo.telemetry_names
+            finally:
+                await salinfo.close()
+
+        # Create clients and listen to ws messages
+        self.log.info(
+            f"Waiting for {self.config.number_of_clients} Manager Clients to be ready"
+        )
+        loop = asyncio.get_event_loop()
```

Comment:
Are you sure you need to do this? Can you use `asyncio.create_task` (in line 294) instead? I think it does the exact same thing.

Reply:
I agree. Furthermore, you should basically never use `asyncio.get_event_loop()` as it has unpredictable behavior (it will try to create a loop if none is running, which can cause hard-to-diagnose errors). The preferred alternative is `asyncio.get_running_loop()` when you need a loop, which is rare (my main use is `loop.run_in_executor`)